src/nvector/hip/nvector_hip.hip.cpp

```diff
@@ -1947,11 +1953,11 @@ int N_VLinearCombinationVectorArray_Hip(int nvec, int nsum, sunrealtype* c,
 
   // Free device arrays
   err = hipFree(d_c);
-  if (!SUNDIALS_HIP_VERIFY(err)) { return (-1); }
+  if (!SUNDIALS_HIP_VERIFY(err)) { return SUN_ERR_GENERIC; }
   err = hipFree(d_Xd);
-  if (!SUNDIALS_HIP_VERIFY(err)) { return (-1); }
+  if (!SUNDIALS_HIP_VERIFY(err)) { return SUN_ERR_GENERIC; }
   err = hipFree(d_Zd);
-  if (!SUNDIALS_HIP_VERIFY(err)) { return (-1); }
+  if (!SUNDIALS_HIP_VERIFY(err)) { return SUN_ERR_GENERIC; }
 
   return hipGetLastError();
```

Comment:
Looks like this is the only function that retured `hipGetLastError`. Everywhere else `SUN_SUCCESS` is returned.
```suggestion
  return SUN_SUCCESS;
```